compiler/optimizer/OMRValuePropagation.hpp

```diff
@@ -80,6 +80,10 @@ extern const ValuePropagationPtr constraintHandlers[];
 typedef TR::typed_allocator<std::pair<TR::CFGEdge * const, TR_BitVector*>, TR::Region &> DefinedOnAllPathsMapAllocator;
 typedef std::map<TR::CFGEdge *, TR_BitVector *, std::less<TR::CFGEdge *>, DefinedOnAllPathsMapAllocator> DefinedOnAllPathsMap;
 
+typedef TR::typed_allocator<std::pair<TR::Node * const, List<TR_Pair<TR::TreeTop, TR::CFGEdge>> *>, TR::Region &> CallNodeToGuardNodesMapAllocator;
+typedef std::map<TR::Node * const, List<TR_Pair<TR::TreeTop, TR::CFGEdge>> *, std::less<TR::Node *>, CallNodeToGuardNodesMapAllocator> CallNodeToGuardNodesMap;
```

Comment:
Just curious why the lists are by pointer instead of by value

Reply:
I was getting some compilation errors. But let me try again...

Reply:
Yeah, it complains about List::List() being protected. It worked with TR_Array because TR_Array simply zeros all elements and does not call element constructor.

Reply:
Left as List pointer..

Reply:
I suspect you still might be able to use it if you use `insert` instead of `operator[]`, but it's not a huge deal :shrug:

Reply:
Oh, ic! I will try to improve later.